pkg/scheduler/api/resource_info.go

```diff
@@ -68,32 +93,44 @@ func (r *Resource) Clone() *Resource {
 	return clone
 }
 
-var minResource float64 = 0.1
+// String returns resource details in string format
+func (r *Resource) String() string {
+	str := fmt.Sprintf("cpu %0.2f, memory %0.2f", r.MilliCPU, r.Memory)
+	for rName, rQuant := range r.ScalarResources {
+		str = fmt.Sprintf("%s, %s %0.2f", str, rName, rQuant)
+	}
+	return str
+}
 
-// NewResource create a new resource object from resource list
-func NewResource(rl v1.ResourceList) *Resource {
-	r := EmptyResource()
-	for rName, rQuant := range rl {
-		switch rName {
-		case v1.ResourceCPU:
-			r.MilliCPU += float64(rQuant.MilliValue())
-		case v1.ResourceMemory:
-			r.Memory += float64(rQuant.Value())
-		case v1.ResourcePods:
-			r.MaxTaskNum += int(rQuant.Value())
-		default:
-			//NOTE: When converting this back to k8s resource, we need record the format as well as / 1000
-			if v1helper.IsScalarResourceName(rName) {
-				r.AddScalar(rName, float64(rQuant.MilliValue()))
-			}
+// ResourceNames returns all resource types
+func (r *Resource) ResourceNames() []v1.ResourceName {
+	resNames := []v1.ResourceName{v1.ResourceCPU, v1.ResourceMemory}
+
+	for rName := range r.ScalarResources {
+		resNames = append(resNames, rName)
+	}
+
+	return resNames
+}
+
+// Get returns the resource value for that particular resource type
+func (r *Resource) Get(rn v1.ResourceName) float64 {
+	switch rn {
+	case v1.ResourceCPU:
+		return r.MilliCPU
+	case v1.ResourceMemory:
+		return r.Memory
+	default:
+		if r.ScalarResources == nil {
+			return 0
 		}
+		return r.ScalarResources[rn]
 	}
-	return r
 }
 
-// IsEmpty returns bool after checking any of resource is less than min possible value
+// IsEmpty returns false if any kind of resource is not less than min value, otherwise returns true
 func (r *Resource) IsEmpty() bool {
-	if !(r.MilliCPU < minResource && r.Memory < minResource) {
+	if r.MilliCPU >= minResource || r.Memory >= minResource {
```

Comment:
do not use '>=' for float comparing.  Combining '!' and '<'  is a general way.

Reply:
> do not use '>=' for float comparing. Combining '!' and '<' is a general way.

Yes, i will recover it